src/modelframe.jl

```diff
@@ -102,34 +104,48 @@ end
 
 const DEFAULT_CONTRASTS = DummyCoding
 
+_levels(x::Union{CategoricalArray, NullableCategoricalArray}) = levels(x)
```

Comment:
Actually that was quite easy. What is going to be harder is to handle DataArrays the same way, since `unique(::PooledDataArray)` does not return values in the order specified by levels. Since `unique` does not guarantee any particular order for levels, I suggest we change DataArrays to follow the ordering of levels (just like `CategoricalArray`). That way, we don't need any common API other than `unique`.